internal/statestore/redis.go

```diff
@@ -535,36 +516,45 @@ func (rb *redisBackend) UpdateAssignments(ctx context.Context, ids []string, ass
 	if assignment == nil {
 		return status.Error(codes.InvalidArgument, "assignment is nil")
 	}
+	if len(ids) == 0 {
+		return nil
+	}
 
 	redisConn, err := rb.connect(ctx)
 	if err != nil {
 		return err
 	}
 	defer handleConnectionClose(&redisConn)
 
-	err = redisConn.Send("MULTI")
+	b := make([]interface{}, len(ids))
+	for i := range ids {
+		b[i] = ids[i]
+	}
+
+	ticketBytes, err := redis.ByteSlices(redisConn.Do("MGET", b...))
 	if err != nil {
 		return err
 	}
 
-	// Sanity check to make sure all inputs ids are valid
-	tickets := []*pb.Ticket{}
-	for _, id := range ids {
-		select {
-		case <-ctx.Done():
-			return ctx.Err()
-		default:
-			var ticket *pb.Ticket
-			ticket, err = rb.GetTicket(ctx, id)
-			if err != nil {
-				redisLogger.WithError(err).Errorf("failed to get ticket %s from redis when updating assignments", id)
-				return err
-			}
-			tickets = append(tickets, ticket)
+	tickets := make([]*pb.Ticket, 0, len(ticketBytes))
+	for i, ticketByte := range ticketBytes {
+		// Tickets may be deleted by the time we read it from redis.
+		if ticketByte == nil {
+			return status.Errorf(codes.NotFound, "failed to get ticket %s from statestore", ids[i])
```

Comment:
Aha. This seems to suggest that MGET will return nil object for missing ticket id? if that is so - ignore and resolve previous comment.

Reply:
yup, the missing objects are nil